File: apps/s-2-admin/src/api/workflow/WorkflowWhereInput.ts
import { IntFilter } from "../../util/IntFilter";
import { StringFilter } from "../../util/StringFilter";
import { UserWhereUniqueInput } from "../user/UserWhereUniqueInput";
import { IntNullableFilter } from "../../util/IntNullableFilter";
import { WorkflowStepListRelationFilter } from "../workflowStep/WorkflowStepListRelationFilter";
import { WorkflowsOnEventTypeListRelationFilter } from "../workflowsOnEventType/WorkflowsOnEventTypeListRelationFilter";

export type WorkflowWhereInput = {
  id?: IntFilter;
  name?: StringFilter;
  user?: UserWhereUniqueInput;
  trigger?: "BEFORE_EVENT" | "EVENT_CANCELLED" | "NEW_EVENT";
  time?: IntNullableFilter;
  timeUnit?: "DAY" | "HOUR" | "MINUTE";
  steps?: WorkflowStepListRelationFilter;
  activeOn?: WorkflowsOnEventTypeListRelationFilter;
};

File: apps/cactus-demo/src/APIs/Workflow/Dtos/WorkflowWhereInput.cs
using CactusDemo.Core.Enums;

namespace CactusDemo.APIs.Dtos;

public class WorkflowWhereInput
{
    public int? Id { get; set; }

    public string? Name { get; set; }

    public TriggerEnum? Trigger { get; set; }

    public int? Time { get; set; }

    public TimeUnitEnum? TimeUnit { get; set; }

    public int? Users { get; set; }

    public List<int>? WorkflowStep { get; set; }

    public List<int>? WorkflowsOnEventTypes { get; set; }
}
